base_decorator.rb

```diff
@@ -0,0 +1,13 @@
+require './nameable'
+
+class Decorator
```

Comment:
You have required `./nameable` but you have not inherited from it. `Decorator` class should be the subclass of `Nameable`. Kindly do so.

Reply:
Hi @muneebulrehman,
I fixed the required changes.